### ISSUE TITLE
UDP will stop listening after timeout

### ISSUE THREAD

**Issue by fortuna** [Jigsaw-Code/outline-go-tun2socks#109]
If a read hits the timeout, we stop listening for packets:
https://github.com/Jigsaw-Code/outline-go-tun2socks/blob/ab0408a6d301b3a55105b94255c1d436c4f6480d/shadowsocks/udp.go#L52-L56

I wonder if this is causing issues with apps that use UDP, specially in extreme conditions.

The default timeout is 30 seconds: https://github.com/Jigsaw-Code/outline-go-tun2socks/blob/2f14928f97f3730e8c5c387003284db8fc12b445/outline/tunnel.go#L84.

It's not clear to me that we need the timeout. Tun2socks stack will supposedly call `udpHandler.Close()` when the app socket is closed.

My suggestion is to remove the timeout. At least make is longer, like the 5 minutes we use on the server:
https://github.com/Jigsaw-Code/outline-ss-server/blob/8269318bb6268f9de5b289c31a59da7617520ce8/server.go#L51

**Comment by bemasc** [comment]
> Tun2socks stack will supposedly call udpHandler.Close() when the app socket is closed.

No, tun2socks has no way to know whether the app socket has been closed.  It only has access to the "tun device" as an io.ReadWriteCloser, with IP packets flowing in and out.
